bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java

```diff
@@ -74,162 +74,166 @@ public JdbcConfiguration(Map<Object, Object> configuration) {
+            logger.debug("No jdbc:user parameter defined in openhab.cfg");
+        }
+        if (StringUtils.isBlank(password)) {
+            logger.debug("No jdbc:password parameter defined in openhab.cfg.");
+        }
+
+        if (StringUtils.isBlank(url)) {
             logger.warn(
-                    "JDBC::updateConfig: SQL user is missing - please configure the jdbc:user parameter in openhab.cfg");
-        } else if (StringUtils.isBlank(password)) {
-            logger.warn("JDBC::updateConfig: SQL password is missing. Attempting to connect without password. "
-                    + "To specify a password configure the jdbc:password parameter in openhab.cfg.");
-        } else if (StringUtils.isBlank(url)) {
-            logger.warn(
-                    "JDBC::updateConfig: SQL url is missing - please configure url in openhab.cfg like 'jdbc:<service>:<host>[:<port>;<attributes>]'");
-        } else if ("false".equalsIgnoreCase(parsedURL.getProperty("parseValid"))) {
+                    "JDBC url is missing - please configure in openhab.cfg like 'jdbc:<service>:<host>[:<port>;<attributes>]'");
+            return;
+        }
+
+        if ("false".equalsIgnoreCase(parsedURL.getProperty("parseValid"))) {
             Enumeration<?> en = parsedURL.propertyNames();
             String enstr = "";
             for (Object key : Collections.list(en)) {
                 enstr += key + " = " + parsedURL.getProperty("" + key) + "\n";
             }
             logger.warn(
-                    "JDBC::updateConfig: SQL url is not well formatted: {}\nPlease configure url in openhab.cfg like 'jdbc:<service>:<host>[:<port>;<attributes>]'",
+                    "JDBC url is not well formatted: {}\nPlease configure in openhab.cfg like 'jdbc:<service>:<host>[:<port>;<attributes>]'",
                     enstr);
-        } else {
-            logger.debug("JDBC::updateConfig: mandatory user={}", user);
-            logger.debug("JDBC::updateConfig: mandatory password=<masked> password.length={}", password.length());
-            logger.debug("JDBC::updateConfig: mandatory url={}", url);
-
-            // set database type and database type class
-            setDBDAOClass(parsedURL.getProperty("dbShortcut")); // derby, h2, hsqldb, mariadb, mysql, postgresql,
-                                                                // sqlite
-            // set user
+            return;
+        }
+
+        logger.debug("JDBC::updateConfig: user={}", user);
+        logger.debug("JDBC::updateConfig: password exists? {}", password != null & !StringUtils.isBlank(password));
+        logger.debug("JDBC::updateConfig: url={}", url);
+
+        // set database type and database type class
+        setDBDAOClass(parsedURL.getProperty("dbShortcut")); // derby, h2, hsqldb, mariadb, mysql, postgresql,
+                                                            // sqlite
+        // set user
+        if (user != null && !StringUtils.isBlank(user)) {
```

Comment:
This can be simplified to `if (StringUtils.isNotBlank(user))`.

https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isNotBlank(java.lang.String)